sw/device/sca/aes_serial.c

```diff
@@ -87,9 +89,45 @@ uint8_t batch_plaintexts[kNumBatchOpsMax][kAesTextLength];
 bool sample_fixed = true;
 
 /**
- * Fixed key for fvsr key TVLA batch capture.
+ * The generation key for TVLA batch capture - specified in [DTR] Section 5.1.
+ * This key is used for generating all pseudo-random data for batch captures.
  */
-uint8_t key_fixed[kAesKeyLength];
+const uint8_t key_gen[kAesKeyLength] = {0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc,
+                                        0xde, 0xf1, 0x23, 0x45, 0x67, 0x89,
+                                        0xab, 0xcd, 0xe0, 0xf0};
+/**
+ * An array to store pre-computed round keys derived from key_gen.
+ */
+uint32_t key_gen_round_keys[(kAesKeyLength / 4) * 11];
+
+/**
+ * Plaintext of the fixed set of fixed-vs-random-key TVLA
+ */
+uint8_t plaintext_fixed[kAesTextLength] = {0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa,
+                                           0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa,
+                                           0xaa, 0xaa, 0xaa, 0xaa};
+/**
+ * Key of the of the fixed set of fixed-vs-random-key TVLA
+ */
+uint8_t key_fixed[kAesTextLength] = {0x81, 0x1E, 0x37, 0x31, 0xB0, 0x12,
+                                     0x0A, 0x78, 0x42, 0x78, 0x1E, 0x22,
+                                     0xB2, 0x5C, 0xDD, 0xF9};
```

Comment:
Also this one is fixed and never changes right? So, it could be `const` as well. Or would this be changing if we would do fvsr data TVLA?

Reply:
If yes, I would leave this as is.

Reply:
> If yes, I would leave this as is.

We would indeed need to use a different constant for fvsr data TVLA. So for now I would keep this as a variable.

Reply:
> These variable should be `static` to limit the scope to this file only.

Thanks for catching this. I've changed it now.